source/slang-capture-replay/slang-session.cpp

```diff
@@ -159,38 +179,102 @@ namespace SlangCapture
         SlangInt conformanceIdOverride,
         ISlangBlob** outDiagnostics)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
-        SlangResult result = m_actualSession->createTypeConformanceComponentType(type, interfaceType, outConformance, conformanceIdOverride, outDiagnostics);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+
+        slang::ITypeConformance* pConformance = nullptr;
+        SlangResult result = m_actualSession->createTypeConformanceComponentType(type, interfaceType, &pConformance, conformanceIdOverride, outDiagnostics);
+
+        TypeConformanceCapture* conformanceCapture = new TypeConformanceCapture(pConformance);
+        Slang::ComPtr<TypeConformanceCapture> resultCapture(conformanceCapture);
+        *outConformance = resultCapture.detach();
+
         return result;
     }
 
     SLANG_NO_THROW SlangResult SessionCapture::createCompileRequest(
         SlangCompileRequest**   outCompileRequest)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         SlangResult result = m_actualSession->createCompileRequest(outCompileRequest);
         return result;
     }
 
     SLANG_NO_THROW SlangInt SessionCapture::getLoadedModuleCount()
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         SlangInt count = m_actualSession->getLoadedModuleCount();
         return count;
     }
 
     SLANG_NO_THROW slang::IModule* SessionCapture::getLoadedModule(SlangInt index)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         slang::IModule* pModule = m_actualSession->getLoadedModule(index);
+
+        if (pModule)
+        {
+            ModuleCapture* moduleCapture = m_mapModuleToCapture.tryGetValue(pModule);
+            if (!moduleCapture)
+            {
+                assert(!"Module not found in mapModuleToCapture");
+            }
+            return static_cast<slang::IModule*>(moduleCapture);
+        }
+
         return pModule;
     }
 
     SLANG_NO_THROW bool SessionCapture::isBinaryModuleUpToDate(const char* modulePath, slang::IBlob* binaryModuleBlob)
     {
-        slangCaptureLog(LogLevel::Verbose, "%s\n", __func__);
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
         bool result = m_actualSession->isBinaryModuleUpToDate(modulePath, binaryModuleBlob);
         return result;
     }
 
+    ModuleCapture* SessionCapture::getModuleCapture(slang::IModule* module)
+    {
+        ModuleCapture* moduleCapture = nullptr;
+        moduleCapture = m_mapModuleToCapture.tryGetValue(module);
+        if (!moduleCapture)
+        {
+            moduleCapture = new ModuleCapture(module);
```

Comment:
Shouldn't we worry about the case where the memory actually ran out because the application used all up?
If that happens, `moduleCapture` will get a null_ptr, and this function will need to handle it properly.

Reply:
If this is the case, then slang will be crash, we cannot do anything about it.
IMO, we shouldn't worry about this extremely corner case.

Reply:
A proper way to handle OOM is to check if the `new` returned null_ptr, and simply return null_ptr.

What could happen from my experience is that the memory is not completely out-of-memory but not enough to allocate for Slang.
From the application developer point of view, Slang is crashing when spawning a certain number of threads with no clear indication.

We should handle the out-of-memory case properly, which is not hard to do in this specific case.